src/app/utils/api/highlights.ts

```diff
@@ -0,0 +1,36 @@
+import { createPostOptions, fetchJSON } from '.';
+import {
+  MatchHighlightClient,
+  PaginatedMatchHighlightsClient,
+} from '../../../types';
+
+export async function postHighlight(
+  highlight: Omit<MatchHighlightClient, '_id' | 'createdAt'>
+): Promise<MatchHighlightClient> {
+  const data = JSON.stringify(highlight);
+  const postOptions = createPostOptions(data);
+
+  return fetchJSON<MatchHighlightClient>('/api/highlights', postOptions);
+}
+
+export async function getHighlights(query: {
+  page: number;
+  itemsPerPage: number;
+}): Promise<PaginatedMatchHighlightsClient> {
+  const { page, itemsPerPage, ...rest } = query;
+  const searchParams = new URLSearchParams({
+    page: page.toString(),
+    itemsPerPage: itemsPerPage.toString(),
+    ...rest,
+  });
+  const queryString = searchParams.toString();
```

Comment:
looks good, is it possible to outsource it to another function, which returns the string directly, you use the same logic in two different endpoints

Reply:
Good point, I will make this change later :-)